crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -1278,6 +1278,22 @@ impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for Paymen
             None => (None, None, None),
         };
 
+        let shipping_cost = payment_data.payment_intent.shipping_cost;
+
+        let pmt_order_tax_amount = payment_data
```

Comment:
here you should check if the payment method type that is passed in the confirm request is the same, only then consider this tax amount